primitives/application-crypto/src/ed25519.rs

```diff
@@ -19,16 +19,35 @@
 use crate::{RuntimePublic, KeyTypeId};
 
 use sp_std::vec::Vec;
+use sp_core::crypto::CryptoTypeId;
 
 pub use sp_core::ed25519::*;
 
+/// An identifier used to match public keys against ed25519 keys
+pub const CRYPTO_ID: CryptoTypeId = CryptoTypeId(*b"ed25");
+
 mod app {
+    use sp_core::crypto::{CryptoTypePublicPair, Public as TraitPublic};
 	use sp_core::testing::ED25519;
+    use crate::ed25519::CRYPTO_ID;
+
 	crate::app_crypto!(super, ED25519);
 
 	impl crate::traits::BoundToRuntimeAppPublic for Public {
 		type Public = Self;
 	}
+
+    impl From<Public> for CryptoTypePublicPair {
+        fn from(key: Public) -> Self {
+            (&key).into()
```

Comment:
Nitpick: isn't there a way to avoid copying here? I.e. extract raw vec directly from `Public`?

Reply:
hmm, i am not sure i see the "copying" here. The purpose of this line is to call `from` from this block:
```
impl From<&Public> for CryptoTypePublicPair {
```
Which is below the one we're commenting on. Can you elabore on where the copy happens?